src/Native/Runtime/unix/PalRedhawkUnix.cpp

```diff
@@ -506,6 +509,10 @@ extern "C" void PalAttachThread(void* thread)
 extern "C" bool PalDetachThread(void* thread)
 {
     UNREFERENCED_PARAMETER(thread);
+    if (g_threadDetachCallback != nullptr)
```

Comment:
I think that using a callback is a reasonable option here but I agree with @jkotas that it should definitely go directly from CoreLib to MRT.